File: src/main/java/Starter.java
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.lang.reflect.InvocationTargetException;
import java.util.Scanner;

public class Starter {

    private Starter() {
        //TODO: dynamic timing testing with p 0 and j 0

        String cmd;
        Scanner scanner = new Scanner(System.in);
        System.out.println("CLI active...");
        while (!(cmd = scanner.nextLine()).equals("exit")) {
            String[] command = cmd.split(" ");

            if (command.length != 2) {
                System.err.println("Invalid number of arguments!");
                continue;
            }

            try {
                int problemNumber = Integer.parseInt(command[1]);
                System.out.println("\033[4mProblem " + problemNumber + "\n\033[0m");

                if (problemNumber > 0) {
                    try {
                        Document doc = Jsoup.connect("https://projecteuler.net/problem=" + problemNumber).get();
                        System.out.println(doc.select("div.problem_content").first().text().replace(". ", ".\n"));
                        System.out.println();
                    } catch (IOException e) {
                        System.err.println("Can't find problem description!");
                    }
                }

                switch (command[0].toLowerCase()) {
                    case "py":
                    case "python":
                    case "p":
                        solvePython(problemNumber);
                        break;
                    case "jar":
                    case "java":
                    case "j":
                        solveJava(problemNumber);
                        break;
//                    case "c#":
//                    case "csharp":
//                    case "c":
//                        break;
                    default:
                        System.out.println("Invalid language: " + command[0]);
                        break;
                }

            } catch (NumberFormatException e) {
                System.err.println("Invalid argument!");
            }
        }

        System.out.println("CLI closed!");
    }

    private static void solvePython(int problemNumber) {
        try {
            ProcessBuilder builder = new ProcessBuilder("cmd.exe", "/c", "cd src && cd main && cd python && starter.py " + problemNumber);
            builder.redirectErrorStream(true);
            Process p = builder.start();
            BufferedReader br = new BufferedReader(new InputStreamReader(p.getInputStream()));

            String line;
            long start_time = System.nanoTime();
            while ((line = br.readLine()) != null) {
                if (line.contains("SOLUTION:")) {
                    String solution = line.split(":")[1];

                    long end_time = System.nanoTime() - start_time;
                    printResult(solution, end_time - 160653100);
                    return;
                }
                System.out.println(line);
            }

            System.err.println("Problem not solved yet!");

        } catch (IOException e) {
            System.err.println("Problem not solved yet!");
        }
    }

    private static void solveJava(int problemNumber) {
        try {
            StringBuilder pack = new StringBuilder("p" + (problemNumber / 100));
            while (pack.length() < 3) pack.insert(1, "0");

            StringBuilder pN = new StringBuilder("" + problemNumber);
            while (pN.length() < 4) pN.insert(0, "0");

            long start_time = System.nanoTime();
            Object solution = Class.forName(pack + ".Problem_" + pN).getDeclaredConstructor().newInstance().getClass().getMethod("solve").invoke(null);
            long end_time = System.nanoTime() - start_time;

            printResult(solution, end_time - 30500);

        } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InvocationTargetException | InstantiationException e) {
            System.err.println("Problem not solved yet!");
        }
    }

    private static void printResult(Object solution, long time) {
        StringBuilder t = new StringBuilder(time + "");
        while (t.length() < 11) t.insert(0, 0);
        for (int i = t.length() - 3; i > 0; i -= 3)
            t = new StringBuilder(t.substring(0, i) + ":" + t.substring(i));

        System.out.println("Solution: " + solution);
        System.out.println("Time:     " + t.toString());
        System.out.println("         [ s| ms| μs| ns]");
    }


    public static void main(String[] args) {
        new Starter();
    }
}

File: src/main/csharp/Starter.cs
using System;
using System.Diagnostics;
using System.Globalization;
using System.Linq;
using System.Reflection;
using System.Text.RegularExpressions;
using System.Threading;
using HtmlAgilityPack;


namespace ProjectEuler.main.csharp
{
    public class Starter
    {
        private static void Main(string[] args)
        {
            Thread.CurrentThread.CurrentCulture = CultureInfo.CreateSpecificCulture("en-US");
            Console.WriteLine("\nC# CLI started...");
            while (true)
                try
                {    
                    Console.Write("=> ");
                    var input = Console.ReadLine();
                    if (input.Equals("exit"))
                    {
                        Console.WriteLine("\nC# CLI closed!");
                        return;
                    }

                    try
                    {
                        Console.WriteLine("--Problem " + input + "--\n");
                        Console.WriteLine(Regex.Replace(new HtmlWeb().Load("https://projecteuler.net/problem=" + input).DocumentNode.SelectNodes("//div[@class='problem_content']")[0]
                            .InnerText.Replace(". ", ".\n"), @"^\s+$[\r\n]*", string.Empty, RegexOptions.Multiline));
                    }
                    catch (Exception)
                    {
                        ErrPrint("Can't find problem description!");
                        continue;
                    }

                    //start Problems:
                    var package = Convert.ToString(Convert.ToInt32(input) / 100);
                    var path = "ProjectEuler.main.csharp.p" + new string('0', 2 - package.Length) + package + ".Problem" + new string('0', 4 - input.Length) + input;
                    var problem = Type.GetType(path);

                    if (problem != null)
                    {
                        var problemObject = Activator.CreateInstance(problem);
                        var toInvoke = problemObject.GetType().GetMethod("Solve", BindingFlags.Static | BindingFlags.Public);
                        if (toInvoke != null)
                        {
                            var startTime = GetNanoseconds();
                            var result = toInvoke.Invoke(null, null);
                            var totalTime = GetNanoseconds() - startTime;

                            var time = totalTime + "";
                            while (time.Length < 11) time = "0" + time;
                            for (var i = time.Length - 3; i > 0; i -= 3) time = time.Substring(0, i) + ":" + time.Substring(i);

                            Console.WriteLine("Solution: " + result);
                            Console.WriteLine("Time:     " + time);
                            Console.WriteLine("         [ s| ms| μs| ns]");
                        }
                    }
                    else
                    {
                        ErrPrint("Problem not solved yet!");
                    }
                }
                catch (Exception e)
                {
                    Console.WriteLine(e + "\n");
                    ErrPrint("Problem not solved yet!");
                }
        }

        private static void ErrPrint(string message)
        {
            Console.ForegroundColor = ConsoleColor.Red;
            Console.WriteLine(message);
            Console.ResetColor();
        }

        private static long GetNanoseconds()
        {
            var timestamp = Stopwatch.GetTimestamp();
            var nanoseconds = 1_000_000_000.0 * timestamp / Stopwatch.Frequency;

            return (long) nanoseconds;
        }
    }
}